src/storage/src/hummock/compactor.rs

```diff
@@ -184,25 +190,47 @@ impl Compactor {
 
     /// Handle a compaction task and report its status to hummock manager.
     /// Always return `Ok` and let hummock manager handle errors.
-    pub async fn compact(context: Arc<CompactorContext>, compact_task: CompactTask) -> bool {
+    pub async fn compact(context: Arc<CompactorContext>, mut compact_task: CompactTask) -> bool {
         tracing::info!(
             "Ready to handle compaction task: \n{}",
             compact_task_to_string(&compact_task)
         );
         let start_time = Instant::now();
 
+        compact_task.splits = vec![risingwave_pb::hummock::KeyRange {
```

Comment:
Will we still support sub compaction?

Reply:
It means that we do not support sub compaction temporarily after this PR?

Reply:
> Will we still support sub compaction?

If we do, compaction will generate too many SSTs

Reply:
> It means that we do not support sub compaction temporarily after this PR?

yes, or compaction will generate too many SSTs

Reply:
> Could we judge whether `vnode2unit` is empty before? For example:
> 
> ```rust
> if  !compact_task
>                 .vnode_mappings.is_empty() {
>     compact_task.splits = vec![risingwave_pb::hummock::KeyRange {
>             left: vec![],
>             right: vec![],
>             inf: false,
>         }];
> }
> ```

fixed